dynamic/message_factory.go

```diff
@@ -53,6 +76,11 @@ func NewMessageFactoryWithRegistries(er *ExtensionRegistry, ktr *KnownTypeRegist
 	}
 }
 
+func (f *MessageFactory) WithResolver(resolver *protoregistry.Types) *MessageFactory {
+	f.types = &RegistryMessageFactory{types: resolver}
+	return f
+}
```

Comment:
I wasnt sure if I should update `NewMessageFactoryWithRegistries` to have an additional arg. I figure that would break existing usages of that function for users so I opted for this. Let me know what you would prefer.

Reply:
I don't think we want or need this change to leak into the API of this package.

In protoprint, you want to use `*protoregistry.Types` and `dynamicpb.Message` _instead of_ `dynamic.MessageFactory` and `dynamic.Message`. There should be no need to mix.